src/main/java/io/vertx/core/spi/cluster/ClusterManager.java

```diff
@@ -83,20 +75,56 @@
    */
   void nodeListener(NodeListener listener);
 
+  /**
+   * Store the details about this clustered node.
+   */
+  Future<Void> setNodeInfo(NodeInfo nodeInfo);
+
+  /**
+   * Get details about this clustered node.
+   */
+  NodeInfo getNodeInfo();
+
+  /**
+   * Get details about a clustered Vert.x node.
+   *
+   * @param nodeId the clustered node id
+   */
+  Future<NodeInfo> getNodeInfo(String nodeId);
+
   /**
    * Join the cluster
    */
-  void join(Handler<AsyncResult<Void>> resultHandler);
+  Future<Void> join();
 
   /**
    * Leave the cluster
    */
-  void leave(Handler<AsyncResult<Void>> resultHandler);
+  Future<Void> leave();
 
   /**
    * Is the cluster manager active?
    *
-   * @return  true if active, false otherwise
+   * @return true if active, false otherwise
    */
   boolean isActive();
+
+  /**
+   * Share a new messaging handler registration with other nodes in the cluster.
+   */
+  Future<Void> register(String address, RegistrationInfo registrationInfo);
```

Comment:
instead pass `Promise<Void>` so the `ClusterManager` does not need to care about which context to use

Reply:
done